lib/Search/DeckProvider.php

```diff
@@ -77,37 +63,34 @@ public function getName(): string {
 	}
 
 	public function search(IUser $user, ISearchQuery $query): SearchResult {
-		$boards = $this->boardService->getUserBoards();
-
-		$matchedBoards = array_filter($this->boardService->getUserBoards(), static function (Board $board) use ($query) {
-			return mb_stripos($board->getTitle(), $query->getTerm()) > -1;
-		});
-
-		$matchedCards = $this->cardMapper->search(array_map(static function (Board $board) {
-			return $board->getId();
-		}, $boards), $query->getTerm(), $query->getLimit(), $query->getCursor());
-
-		$self = $this;
+		$cursor = $query->getCursor() !== null ? (int)$query->getCursor() : null;
+		$boardResults = $this->searchService->searchBoards($query->getTerm(), $query->getLimit(), $cursor);
+		$cardResults = $this->searchService->searchCards($query->getTerm(), $query->getLimit(), $cursor);
 		$results = array_merge(
 			array_map(function (Board $board) {
 				return new BoardSearchResultEntry($board, $this->urlGenerator);
-			}, $matchedBoards),
-
-			array_map(function (Card $card) use ($self) {
-				$board = $self->boardService->find($self->cardMapper->findBoardId($card->getId()));
-				$stack = $self->stackMapper->find($card->getStackId());
-				return new CardSearchResultEntry($board, $stack, $card, $this->urlGenerator);
-			}, $matchedCards)
+			}, $boardResults),
+			array_map(function (Card $card) {
+				return new CardSearchResultEntry($card->getRelatedBoard(), $card->getRelatedStack(), $card, $this->urlGenerator);
+			}, $cardResults)
 		);
 
-		return SearchResult::complete(
+		if (count($cardResults) < $query->getLimit()) {
+			return SearchResult::complete(
+				'Deck',
```

Comment:
Translate?

Reply:
I guess that is a different discussion, as we currently never translate the app name for deck.